examples/nlp/scripts/process_wiki_zh.py

```diff
@@ -0,0 +1,155 @@
+                                  "[MASK]", "[UNK]")):
+    """Create vocabulary from lines"""
+    # Count word occurency
+    vocab = {}
+    for line in lines:
+        if line.strip():
+            for w in line.strip().split():
+                if w in vocab:
+                    vocab[w] += 1
+                else:
+                    vocab[w] = 1
+
+    # Remove rare words
+    for w in vocab.keys():
+        if vocab[w] < min_frequency:
+            del vocab[w]
+            print("Remove rare word {} from vocab".format(w))
+
+    # Include special symbols and write to file
+    vocab = list(special_symbols) + sorted(vocab.keys())
+    with open(vocab_file, 'w') as f:
+        for w in vocab:
+            f.write(w + '\n')
+    print(f"Created vocabulary of size {len(vocab)}")
+
+    return len(vocab)
+
+
+def read(filename, regex):
+    """Read file, filter by regex, add space between characters
+    """
+    with open(filename, encoding='utf-8') as f:
+        documents = f.readlines()
+
+    r = re.compile(regex)
+    # Each line is a full article
+    for i, line in enumerate(documents):
+        text = json.loads(line)['text']
+        text = text.replace('\\n', '\n').strip()
+        text = r.findall(text)
+        text = ' '.join([c for c in text])  # separate char by space
+        documents[i] = text
+
+    return documents
+
+
+def read_files(data_dir, regex, max_dirs=-1):
+    """Read text in all sub-directories, filter by regex and return lines"""
+    executor = ProcessPoolExecutor(max_workers=4)
+
+    tasks = []
+    sub_dirs = [name for name in os.listdir(data_dir)
+                if os.path.isdir(os.path.join(data_dir, name))]
+    for sub_dir in sub_dirs[:max_dirs]:
+        p_dir = os.path.join(data_dir, sub_dir)
+        print("To process files in dir: "+p_dir)
+        files = os.listdir(p_dir)
+        for f in files:
+            fullpath = os.path.join(data_dir, sub_dir, f)
+            tasks.append(executor.submit(partial(read, fullpath, regex)))
+    print('Preprocessing texts, please wait...', flush=True)
+
+    # Collect text
+    lines = []
+    for t in tasks:
+        for doc in t.result():
+            splitted = re.split(r'[\n]', doc)
+            lines += splitted
+
+    return lines
+
+
+def save(output_dir, lines, train_p=0.95):
+    """Save lines into two files within output dir: train.txt and val.txt"""
+    # Take large portino as training set
+    train_file = os.path.join(output_dir, 'train.txt')
+    num_train_doc = int(train_p * len(lines))
+    with open(train_file, 'w', encoding='utf-8') as f:
+        for l in lines[:num_train_doc]:
+            f.write(l+'\n')
+
+    # Take remainings as validation set
+    val_file = os.path.join(output_dir, 'valid.txt')
+    with open(val_file, 'w', encoding='utf-8') as f:
+        for l in lines[num_train_doc:]:
+            f.write(l+'\n')
+
+    print("Finished processing.")
+    print("Processed text saved to {} and {}".format(train_file, val_file))
```

Comment:
f-string + logger

Reply:
Thanks! Used f-string, but no logger as we don't import nemo in this script.